pkg/controller/manage/kmesh_manage.go

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright 2024 The Kmesh Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kmeshmanage
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	netns "github.com/containernetworking/plugins/pkg/ns"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+
+	"kmesh.net/kmesh/pkg/constants"
+	ns "kmesh.net/kmesh/pkg/controller/netns"
+	"kmesh.net/kmesh/pkg/logger"
+	"kmesh.net/kmesh/pkg/nets"
+)
+
+var (
+	log                = logger.NewLoggerField("kmesh_manage")
+	annotationDelPatch = []byte(fmt.Sprintf(
+		`{"metadata":{"annotations":{"%s":null}}}`,
+		KmeshRedirectionAnnotation,
+	))
+	annotationAddPatch = []byte(fmt.Sprintf(
+		`{"metadata":{"annotations":{"%s":"%s"}}}`,
+		KmeshRedirectionAnnotation,
+		"enabled",
+	))
+)
+
+const (
+	DefaultInformerSyncPeriod  = 30 * time.Second
+	KmeshRedirectionAnnotation = "kmesh.net/redirection"
+)
+
+func NewKmeshManageController(client kubernetes.Interface) (*KmeshManageController, error) {
+	stopChan := make(chan struct{})
+	nodeName := os.Getenv("NODE_NAME")
+
+	informerFactory := informers.NewSharedInformerFactoryWithOptions(client, DefaultInformerSyncPeriod,
+		informers.WithTweakListOptions(func(options *metav1.ListOptions) {
+			options.FieldSelector = fmt.Sprintf("spec.nodeName=%s", nodeName)
+		}))
+
+	podInformer := informerFactory.Core().V1().Pods().Informer()
+
+	if _, err := podInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			pod, ok := obj.(*corev1.Pod)
+			if !ok {
+				log.Errorf("expected *corev1.Pod but got %T", obj)
+				return
+			}
+
+			if !checkValid(pod) {
+				return
+			}
+
+			log.Infof("%s/%s: enable Kmesh manage", pod.GetNamespace(), pod.GetName())
+
+			nspath, _ := ns.GetPodNSpath(pod)
+
+			if err := handleKmeshManage(nspath, true); err != nil {
+				log.Errorf("failed to enable Kmesh manage")
+				return
+			}
+			if err := addKmeshAnnotation(client, pod); err != nil {
+				log.Errorf("failed to add Kmesh annotation, err is %v", err)
```

Comment:
For this failure, it could not recover, we can add a workqueue for better resilience

Reply:
> For this failure, it could not recover, we can add a workqueue for better resilience

yes, added